src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -941,6 +941,18 @@ private function step_in_body() {
 				$this->reconstruct_active_formatting_elements();
 				$this->insert_html_element( $this->state->current_token );
 				return true;
+
+			/*
+			 * > A start tag whose tag name is "input"
+			 */
+			case '+INPUT':
+				$this->reconstruct_active_formatting_elements();
+				$this->insert_html_element( $this->state->current_token );
+				$type_attribute = $this->get_attribute( 'type' );
+				if ( null === $type_attribute || 'hidden' === strtolower( $type_attribute ) ) {
```

Comment:
this could also be `true`, so we need something more like

```php
if ( is_string( $type_attribute ) && 'hidden' === strtolower( $type_attribute ) ) {
	$this->state->frameset_ok = false;
}
```

Reply:
Thanks, I had implemented this incorrectly. I've updated the test I used in the description and included a relevant comment quoted from the standard here.